### DISCUSSION TITLE
API not found for Downloading Images from the DB

### DISCUSSION THREAD

**Discussion by HarisMohre** [exadel-inc/CompreFace#590] [Q&A] [answered]
Hello,

First of. Amazing API this is , especially the UI and the number of API`s . I am actually trying to download an image in Binary Format ofcourse by using the below API call.

curl -X GET "http://localhost:8000/api/v1/recognition/faces/<image_id>/img"
-H "x-api-key: <service_api_key>"

But i keep getting the below error:

{
  timestamp: '2021-08-31T13:08:05.729+00:00',
  status: 404,
  error: 'Not Found',
  message: '',
  path: '/api/v1/recognition/faces/f578bf59-e508-4c25-b238-5fc3a52623e6/img'
}

Is this API discontinued?

Thanks,
Harry.

**Comment by pospielov** [comment]
This is definitely not discontinued, as we released it two days ago :)
First of all, make sure you use the 0.6.0 version.
Second, make sure you didn't mix up the API key and image id.
If nothing helps, could you post the exact request you do

**Reply by HarisMohre** [reply]
Hi Sehrii,

Yeah I had a hunch that I might be doing something wrong here. I am indeed using the **0.6.x** Version and below is my API call:

**curl -X GET "http://localhost:8000/api/v1/recognition/faces/b7b3ea02-0dcb-4ad7-b2b8-c2a3b2aca0db/img"
-H "x-api-key: 00000000-0000-0000-0000-000000000002"**

Thanks,
Haris.

**Reply by pospielov** [reply]
The request looks good, I just run it, and even though I don't have an image with such id, I didn't get the error you have.
Probably the problem can be with the services, could you run `docker ps` and show me the result?

**Reply by HarisMohre** [reply]
here you go sir:

![image](https://user-images.githubusercontent.com/89574723/131633078-bd60884c-3e2b-49c5-9609-58a115d29b70.png)

**Reply by pospielov** [reply]
![image](https://user-images.githubusercontent.com/3736126/131634402-2975a1ef-5e9f-4d45-969c-7c0b0efcd21c.png)
Yeah, I think something is wrong with your images. I mean if you look at my images - they are `exadel/compreface-fe:0.6.0`, and yours are something unknown.
If you had 0.5.1 version before, it can be those images (even thought I don't know why they are unknown now).
you can check my theory my opening UI and checking the version:
![image](https://user-images.githubusercontent.com/3736126/131634904-93be84f6-658c-4b23-9745-b6983c68a227.png)

What I suggest is to try to stop CompreFace and run it again.
Just to be sure your data won't be lost, you need to run the CompreFace from the same folder you did it the first time, so just copy new `docker-compose.yml` and `.env` files into the old directory(or if you deleted it, just rename the current one, so it will be like old one).

if nothing helps, remove those containers manually:
`docker stop compreface-ui`
`docker rm compreface-ui`
(and do it for each container)
Then run CompreFace again

**Reply by HarisMohre** [reply]
ahhhhh.... I see. You were right. It does point to 0.5 Version for some reason. Guess I need to clean up Docker and boot it up again so i can get 0.6 Version. Thanks Mate. for the help. Will update you if this works.

**Reply by HarisMohre** [reply]
Yup, It was the issue with the Image. I downloaded the Master copy and refreshed the Docker Images and containers and now it works :) . Thanks Serhii For the awesome help.

**Comment by HarisMohre** [comment, marked as answer]
Yup, It was the issue with the Image. I downloaded the Master copy and refreshed the Docker Images and containers and now it works :) . Thanks Serhii For the awesome help.

The Issue was that the Docker was still pointing to ver 0.5 instead of 0.6. So a full refresh and loading from Master Resolved the issue.

**Reply by mczarsty** [reply]
hi @HarisMohre ! i am one of the designers for CompreFace. i am currently working on a ux survey - i'd love to chat quickly about your thoughts on where we could potentially improve CompreFace, if you are interested? if you are i can set up a skype or whatsapp chat :)
